benchmark-framework/src/main/java/io/openmessaging/benchmark/worker/DistributedWorkersEnsemble.java

```diff
@@ -144,7 +144,9 @@ public void createProducers(List<String> topics) {
     @Override
     public void startLoad(ProducerWorkAssignment producerWorkAssignment) throws IOException {
         // Reduce the publish rate across all the brokers
-        producerWorkAssignment.publishRate /= numberOfUsedProducerWorkers;
+        if (numberOfUsedProducerWorkers != 0) {
+            producerWorkAssignment.publishRate /= numberOfUsedProducerWorkers;
```

Comment:
Defaults to 0 somewhere else I assume (or I guess that's just java default).

Reply:
The rate? It is read also from the config and could be non-zero, but if there are no producers the rate doesn't matter as no producers are created and it is effectively ignored. The "rate limiter" is always created however, and that's what throws the exception on NaN.

That's actually an interesting difference between N producers on 1 host, vs 1 producer on N hosts: in the former case the rate limiter is shared between all of the producers, so if one producer stops, the others pick up the slack and try to keep the rate. In the 1 producer on N hosts, the rate is statically divided, so if one producer stops the rate will drop by that producer's fraction.